### ISSUE TITLE
issues about visualization

### ISSUE THREAD

**Issue by captain0305** [deeplearning-wisc/stud#4]
Hi! 

I have some problem with visualization using model [model_final_resnet_bdd](https://drive.google.com/file/d/1JAthrSJ8yK5bbhlZAD2vVor2uJ5aTkaX/view?usp=sharing).

using [code](https://github.com/captain0305/captain0305/blob/master/inference_using_defaults.py)

![image](https://user-images.githubusercontent.com/63387697/177728698-b1e99af6-f004-4b50-aba9-78b1986a2d68.png)

 Results are showing the det boxes on the img of all id classes. But there's no ood boxes showing on the img. How could I tell  apart the id and old object using this repo?

Looking forward to your reply!

**Comment by captain0305** [comment]
I have solve this issue with  [this code](https://github.com/captain0305/captain0305/blob/master/inference_using_defaults_new.py)

but I still have some question with this issue. 
```python
inter_feat = results.inter_feat[0:max_boxes]
    print(inter_feat)
    print(torch.logsumexp(inter_feat[:, :-1], dim=1).cpu().data.numpy())
    print((np.argwhere(
            torch.logsumexp(inter_feat[:, :-1], dim=1).cpu().data.numpy() < energy_threshold)).reshape(-1))
    if energy_threshold:
        labels[(np.argwhere(
            torch.logsumexp(inter_feat[:, :-1], dim=1).cpu().data.numpy() < energy_threshold)).reshape(-1)] = 8
```

What is the inter_feat? It has 11 dimension. And after calculating , it is being used to tell apart the ood and id obj. 
Is this just match the step in the paper below?
<img width="548" alt="image" src="https://user-images.githubusercontent.com/63387697/178749833-1a35a85a-13aa-478e-8b72-334317811125.png">
And I wonder why inter_feat has 11 dimension and the meaning of each dimension. And also why the pred_cls_probs has 11 dimension? I just using the model model_final_resnet_bdd, and it's config just have 8 categories. I just quite confused about that.

Looking forward  to your reply

**Comment by d12306** [comment]
hi, @captain0305 , sorry for the late reply! i am really busy recently with experiments. 

1) yes, the inter_feat here is the logits. 

2) actually bdd100k has 10 classes plus a background class. here we use the 11-bit logit for training but essentially we only train with 8 classes. this is the legacy code used in https://github.com/xinw1012/cycle-confusion. you can definitely use 8-bit. i suppose the result to be similar.

**Comment by d12306** [comment]
@captain0305 , feel free to post more questions here if needed! i will check more frequently from now on

**Comment by captain0305** [comment]
Thank u for your reply!
I have trained my data using stud, and have some benchmarks.

**Comment by BigBuffa1o** [comment]
@captain0305
Hello,by applying your code on a single test img,should I modify the original yaml file to get a successful output like you?
